src/internal/consistenthashing/etcd_test.go

```diff
@@ -0,0 +1,294 @@
+package consistenthashing
+
+import (
+	"context"
+	"path"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/collection"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testetcd"
+	"github.com/sirupsen/logrus"
+
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type testRingConfig struct {
+	client *etcd.Client
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+type lockTestConfig struct {
+	ctx            context.Context
+	cancel         context.CancelFunc
+	eg             *errgroup.Group
+	locks          int
+	keys           *sync.Map
+	workers        int
+	workerIds      map[int]string
+	workersReady   chan bool
+	ringConfig     testRingConfig
+	beginLocking   chan bool
```

Comment:
It is pretty common to use empty structs rather than booleans for a use case like this.